### PR TITLE
[jvm-packages] fix early stopping doesn't work even without custom_eval setting

### PR DESCRIPTION
**Pull request by wbo4958**
credit to @sammynammari . Pleas check https://github.com/dmlc/xgboost/issues/6657 and https://github.com/dmlc/xgboost/pull/6708

### CODE REVIEW

**Comment by trivialfis** [comment]
@CodingCat Please help to review.

File: jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/ParameterSuite.scala [hunk 1]
```diff
@@ -78,4 +78,26 @@ class ParameterSuite extends FunSuite with PerTest with BeforeAndAfterAll {
...
+
+    val thrown = intercept[IllegalArgumentException] {
+      new XGBoostClassifier(paramMap).fit(trainingDF)
+    }
+
+    assert(thrown.getMessage.contains("custom_eval does not support early stopping"))
+  }
+
+  test("early stopping should work without custom_eval setting") {
+    val paramMap = Map("eta" -> "0.1", "custom_eval" -> null, "silent" -> "1",
```
Comment (CodingCat) [inline jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/ParameterSuite.scala:96]: if it is to test without custom_eval setting, should we remove `"custom_eval" -> null`?

Comment (wbo4958) [reply jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/ParameterSuite.scala]: Done

**Comment by CodingCat** [comment]
merged, thanks!

**Pull request merged**